AstSemantics.md

```diff
@@ -273,39 +275,53 @@ nested. This guarantees that all resulting control flow graphs are well-structur
 
 ## Calls
 
-Direct calls to a function specify the callee by index into a function table.
+Each function has a *signature*, which consists of:
 
-  * `call_direct`: call function directly
+  * Return types, which are a sequence of local types
+  * Argument types, which are a sequence of local types
+
+Note that WebAssembly itself does not support variable-length argument lists.
+C and C++ compilers are expected to implement this functionality by storing
+an array of arguments in a buffer in linear memory and passing a pointer to
+the buffer.
```

Comment:
Those are ABI concerns that don't belong in AstSemantics.md. The main goal here is just to let people know that we haven't forgotten about varargs :-). But I agree that "array" isn't quite right, so I'll fix that.